controllers/drain_controller.go

```diff
@@ -436,6 +524,45 @@ func (dr *DrainReconcile) findNodePoolConfig(ctx context.Context, node *corev1.N
 	}
 }
 
+// enableSriovDevicePlugin change the device plugin label on the requested node to enable
+// if there is a pod still running we will return false
+func (dr *DrainReconcile) enableSriovDevicePlugin(ctx context.Context, node *corev1.Node) (bool, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("enableSriovDevicePlugin():")
+	pods := &corev1.PodList{}
+	err := dr.List(context.Background(), pods, &client.ListOptions{
+		Raw: &metav1.ListOptions{
+			LabelSelector:   "app=sriov-device-plugin",
+			FieldSelector:   "spec.nodeName=" + node.Name,
+			ResourceVersion: "0"},
+	})
+
+	if err != nil {
+		logger.Error(err, "failed to list device plugin pods running on node")
+		return false, err
+	}
+
+	if len(pods.Items) != 0 {
+		log.Log.Info("device plugin pod still terminating on node")
+		return false, nil
+	}
+
+	// TODO:(schseba): not sure if we should wait here for the device plugin to be running
```

Comment:
yes we should, because otherwise the remaining code is 1) hard to read since the actual state is unknown, 2) if it doesn't come up, we probably want to error out

Reply:
The same holds when we request the dp to restart. You probably don't want to be flipping back and forth and not blocking on state because then there will a difference between what's running and what you think is running.